src/jdk.jlink/share/classes/jdk/tools/jlink/internal/runtimelink/JimageDiffGenerator.java

```diff
@@ -112,4 +114,51 @@ public List<ResourceDiff> generateDiff(ImageResource base, ImageResource image)
         return diffs;
     }
 
+    /**
+     * Compare the contents of the two input streams (byte-by-byte).
+     *
+     * @param is1 The first input stream
+     * @param is2 The second input stream
+     * @return {@code true} iff the two streams contain the same number of
+     *         bytes and each byte of the streams are equal. {@code false}
+     *         otherwise.
+     */
+    private boolean compareStreams(InputStream is1, InputStream is2) {
+        byte[] buf1 = new byte[1024];
+        byte[] buf2 = new byte[1024];
+        int bytesRead1, bytesRead2 = 0;
+        try {
+            try (is1; is2) {
+                while ((bytesRead1 = is1.read(buf1)) != -1 &&
+                       (bytesRead2 = is2.read(buf2)) != -1) {
+                    if (bytesRead1 != bytesRead2) {
```

Comment:
This test seems to assume there won't be any short reads. That might be true in practice, but for defensive programming reasons it would be safer to use `readNBytes()` instead of `read()`.